### DISCUSSION TITLE
dx serve from workspace?

### DISCUSSION THREAD

**Discussion by ThomasCartier** [DioxusLabs/dioxus#2936] [Q&A]
Hi,

How would you run the `dx serve` command inside a workspace ?
I have a ws with different crates and a dioxus project is one of them.

So ideally, we should have a way to `dx serve -p my package`, running from the workspace root's directory.
Like we do `cargo build -p my package` from the root directory.

Maybe it exists already, but I am not sure how to activate it.

**Comment by ealmloff** [comment]
We don't have the `-p` shorthand, but you can pass `--package`

**Reply by ThomasCartier** [reply]
Thanks,

The result is still strange.
My workspace doesn't have a root, it's virtual.

From the workspace root, when I do:

```
cargo build --features web -Z unstable-options --artifact-dir target/debug/out/wasm -p myapp
cargo build --features server -Z unstable-options --artifact-dir target/debug/out -p myapp
```

it works flawlessly.

but doing
`dx serve -- --package myapp` leads to
`Error: Cargo Error: no root package?`

I don't want a root package for my workspace because the crates inside are siblings, but that's it.
It only serves as a compilation time saver.

I run v0.5.6 paired with a nightly toolchain (up to date).
So my current solution is not to use any workspace for my dioxus apps, because the cli generates a good bit of files (js, compression..) and I don't want to replicate this. But It would be nice if I could.

**Reply by ThomasCartier** [reply]
I have to pass `-- --package` because otherwise
```
error: unexpected argument '--package' found

  tip: to pass '--package' as a value, use '-- --package'
```

So maybe v0.5.6 didn't receive the updates to make it work I guess.

**Reply by ThomasCartier** [reply]
Same with 0.5.7

**Comment by ThomasCartier** [comment]
Really sorry to bump again, but could anyone run the d`x build/serve` command in a virtual workspace (no package section in the workspace manifest) ?

**Comment by ThomasCartier** [comment]
I *think* that I will retry when 0.6.* stable is out.
Just tried with cli 0.6.2 and the package option was accepted (great) but I had the same issue as the other user in the v0.6 megathread (index.html not found).

@ealmloff : Thanks for the `-p `option :+1:
